### DISCUSSION TITLE
Салют, а когда уже будет можно писать сценарии на JS?

### DISCUSSION THREAD

**Discussion by awinogradov** [sberdevices/salute-issues#180] [Canvas App]
Кажется, что очень много есть готового тулинга для ботов по написания сценариев на JS. Да и самих разработчиков, которые привыкли всё делать на JS. Может есть какие-то планы в этом направлении?

**Comment by salute-assistant** [comment]
Наши программисты изо всех трудятся над этим. Ждать осталось совсем чуть-чуть!

**Reply by awinogradov** [reply]
Круто!

**Reply by snyuryev** [reply]
Это будет реализовано в редакторе Code?

**Reply by awinogradov** [reply]
Это будет реализовано в виде SDK для NodeJS. Код сценария можно будет запускать на любых серверах или в лямбдах. Примерно так же как сейчас разрабатываются сервисы с использованием expressjs.

**Reply by snyuryev** [reply]
А интенты и сущности будут создаваться в студии?

**Reply by salute-assistant** [reply]
Интенты и сущности будут создаваться локально в конфигурационном файле, который будет синхронизироваться в обе стороны с проектом в Code и SmartApp Brain для тренировки модели. Синхронизацию будет осуществлять CLI.

**Reply by snyuryev** [reply]
Это круто!

**Reply by snyuryev** [reply]
Как раз открыл соседний топик про CLI - было очень удобно все делать в IDE.

**Reply by awinogradov** [reply]
Твой топик немного про другое. В данном случае сценарий, написанный с помощью SDK не будет деплоиться в студию. Это будет сценарий, который работает через webhook.

**Reply by snyuryev** [reply]
Понял.
Сначала подумал что вдруг и для Code тоже сразу будет в комплекте)

**Reply by awinogradov** [reply]
🤷🏼‍♂️
